src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/spec/proceed/confirmation/PayImmediatelyConfText.java

```diff
@@ -27,20 +26,28 @@ public Optional<String> generateTextFor(CaseData caseData) {
         LocalDate whenBePaid = Optional.ofNullable(caseData.getRespondToClaimAdmitPartLRspec())
             .map(RespondToClaimAdmitPartLRspec::getWhenWillThisAmountBePaid)
             .orElse(null);
-        String formattedWhenBePaid = formatLocalDate(whenBePaid, DATE);
-        return Optional.of(format(
-            "They must make sure you have the money by %s. "
-                + "Any cheques or transfers should be clear in your account"
-                + "<p>You need to tell us if you’ve settled the claim, for example because the defendant has paid you.</p>"
-                + "<p>You can settle for less than the full claim amount.</p>"
-                + "<p><h3>If you haven’t been paid.</h3></p>"
-                + "<p>If the defendant has not paid you, you can request a County Court Judgment "
-                + "by completing the following form and sending it to the email address below.</p>"
-                + "<p><li><a href=\"%s\" target=\"_blank\">N225 </a>- Ask for judgment on a claim for a specified amount of money</li></ul></p>"
-                + "<p>Email: <a href=\"mailto:contactocmc@justice.gov.uk\">contactocmc@justice.gov.uk</a></p>",
-            formattedWhenBePaid,
-            "https://formfinder.hmctsformfinder.justice.gov.uk/n225-eng.pdf"
-        ));
+
+        String formattedWhenBePaid = whenBePaid != null ? formatLocalDate(whenBePaid, DATE) : null;
+
+        if (formattedWhenBePaid == null) {
+            throw new IllegalStateException("Unable to format the payment date.");
```

Comment:
lets move this throws clause before line 30 and have check on whenBePaid

Reply:
done